nixos/modules/services/networking/ddclient.nix

```diff
@@ -31,8 +31,8 @@ let
   preStart = ''
     install ${configFile} /run/${RuntimeDirectory}/ddclient.conf
     ${lib.optionalString (cfg.configFile == null) (if (cfg.passwordFile != null) then ''
-      password=$(head -n 1 ${cfg.passwordFile})
-      sed -i "s/^password=$/password=$password/" /run/${RuntimeDirectory}/ddclient.conf
+      password=$(printf "%q" "$(head -n 1 "${cfg.passwordFile}")")
+      sed -i "s|^password=$|password=$password|" /run/${RuntimeDirectory}/ddclient.conf
```

Comment:
this leaks the password through the command line of sed, which is readable in /proc. It would be better to make a small python script (or any other language) that does the substitution after reading the password file without using argv.

Reply:
Good point! (But since this PR doesn't make it worse, I think we can merge.)

Reply:
Thank you. Perhaps you or someone else with more insight than me could address this issues in a new PR?